Basket-Donation-Payments/Basket-Donation-Payments/Models/APIManager.m

```diff
@@ -25,35 +30,44 @@ -(void)fetchBraintreeClientToken {
     NSURL *clientTokenURL = [NSURL URLWithString:@"https://braintree-sample-merchant.herokuapp.com/client_token"];
     NSMutableURLRequest *clientTokenRequest = [NSMutableURLRequest requestWithURL:clientTokenURL];
     [clientTokenRequest setValue:@"text/plain" forHTTPHeaderField:@"Accept"];
-
+    
     [[[NSURLSession sharedSession] dataTaskWithRequest:clientTokenRequest completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
         // TODO: Handle errors
         //NSString *clientToken = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
         NSString *clientToken = @"CLIENT_TOKEN_FROM_SERVER"; //TODO: demo client token
-
+        
         // As an example, you may wish to present Drop-in at this point.
         // Continue to the next section to learn more...
     }] resume];
 }
 
 
-+ (void)createPaymentIntentWithBlock:(void (^)(NSError *, NSDictionary *))completion {
++ (void)createPaymentIntentWithBasket:(Basket*)basket totalAmount:(NSNumber*)totalAmount WithBlock:(void (^)(NSError *, NSDictionary *))completion {
     NSString *backendURL = [APIManager getAPISecretKeysDict][@"Backend_Server_Url"];
+    
     // Create a PaymentIntent by calling your server's endpoint.
     NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@create-payment-intent", backendURL]];
+    
+    BasketTransaction *basketTx = [BasketTransaction new]; //
+    basketTx.basketRecipient = basket;
+    basketTx.madeByUser = [User currentUser];
+    basketTx.totalAmount = totalAmount;
+    //TODO: do I need to save basketTx to Parse if I can retrieve from Stripe API?
+    
+    NSMutableArray<NSDictionary*> *arrayOfNonprofits = [NSMutableArray array];
+    for (Nonprofit *np in basket.nonprofits) {
+        NSDictionary* merchantInfo = @{
```

Comment:
This should also be a model instead of a dictionary

Reply:
I'm using arrayOfNonprofits to input into the "json" NSDictionary that eventually gets passed into the POST request. It needs to be an array of dictionaries, so I can re-access the data in my server-side code, right?

Reply:
Oh that's fine then, sorry for the misunderstanding!